classes/local/fields/learner_profile_field.php

```diff
@@ -0,0 +1,227 @@
+            $classname= 'profile_field_' . $fieldrecord->datatype;
+            /** @var profile_field_base $fieldobject */
+            $field = new $classname($fieldrecord->id);
+            $this->fieldinputname = $field->inputname;
+            return $field;
+        } catch (\coding_exception $exception) {
+            throw $exception;
+        }
+    }
+
+    /**
+     * Build field class based off extra configuration and load user data into class.
+     *
+     * @param stdClass $user
+     * @return mixed|\profile_field_base $field
+     * @throws \coding_exception
+     * @throws \dml_exception
+     */
+    public function get_field_with_user_data(stdClass $user) : \profile_field_base {
+        global $DB, $CFG;
+        try {
+            $fieldid = $this->get_configuration()->get('extraconfigurationdata')->fieldid ?? null;
+            if (is_null($fieldid)) {
+                throw new \coding_exception('nofieldiddefinedinextraconfig');
+            }
+            $sql = "SELECT uif.*, uic.name AS categoryname, uind.id AS hasuserdata, uind.data, uind.dataformat
+                      FROM {user_info_field} uif
+                 LEFT JOIN {user_info_category} uic ON uic.id = uif.categoryid 
+                 LEFT JOIN {user_info_data} uind ON uind.fieldid = uif.id
+                     WHERE uif.id = :fieldid AND uind.userid = :userid";
+            $fieldrecord = $DB->get_record_sql($sql, ['fieldid' => $fieldid, 'userid' => $user->id]);
+            if (!$fieldrecord) {
+                return $this->get_field();
+            }
+            require_once($CFG->dirroot . '/user/profile/field/' . $fieldrecord->datatype . '/field.class.php');
+            $classname = 'profile_field_' . $fieldrecord->datatype;
+            /** @var profile_field_base $fieldobject */
+            $field = new $classname($fieldrecord->id, $user->id, $fieldrecord);
+            $this->fieldinputname = $field->inputname;
+            $field->set_category_name($fieldrecord->categoryname);
+            unset($fieldrecord->categoryname);
+            return $field;
+        } catch (\coding_exception $exception) {
+            throw $exception;
+        }
+    }
+
+    public function build_data_cell($user) {
+        $cell = new cell();
+        $cell->class = "cell";
+        if (isset($user->{$this->fieldinputname})) {
+            $cell->plaintextcontent = $user->{$this->fieldinputname}->display_data();
+        } else {
+            $cell->plaintextcontent = '';
+        }
+        return $cell;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function get_data_for_user(stdClass $user): stdClass
+    {
+        $field = $this->get_field_with_user_data($user);
+        if ($field) {
+            $user->{$field->inputname} = $field;
+        }
+        return $user;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function get_data_for_users(user_list $userlist): array
```

Comment:
Where is get_data_for_users() used? Does it need to be added before release?

Reply:
I don't think so, but I wil sort.